src/main/java/org/ballerinalang/command/cmd/ListCommand.java

```diff
@@ -194,13 +194,10 @@ private static void writeLocalDistsIntoJson(JSONObject distList) {
             if (!file.exists()) {
                 file.getParentFile().mkdirs();
                 file.createNewFile();
+                ToolUtil.addExecutablePermissionToFile(new File(OSUtils.getUserHome() + File.separator +
+                        OSUtils.BALLERINA_HOME_DIR));
                 ToolUtil.addExecutablePermissionToFile(file);
             }
-
-            if (!Files.isWritable(Paths.get(distListFilePath))) {
-                throw ErrorUtil.createCommandException("permission denied: you do not have write access to '" +
```

Comment:
@suganyasuven 
Although we explicitly  set write permissions, couldn't be there any situations user will lose write permissions. Specially in case if they change permissions manually